### DISCUSSION TITLE
PUT/POST SDTM/SEND/ADaM sets of data or individual records (Insert or Update)

### DISCUSSION THREAD

**Discussion by JozefAerts** [cdisc-org/DataExchange-DatasetJson-API#6] [User Stories]
In order to ever come to "really rolling submissions", i.e. that regulatory authorities can already start reviewing even when only a (small) part of the data has been transmitted, we need a mechanism to push data to either the regulatory authority itself or a "neutral zone" between sponsor and regulatory authority.
As such transfers may contain "update" information (e.g. records changed), we need a mechanism (either on the HTTP level or on the dataset/record level) for indicating whether something is "insert", "update" or "delete".
Interesting will be the discussion what the identifier of such a record will be. Will it be the set of natural keys (providing by the "keySequence") or will it be the artificial key (--SEQ)?
All this of course requires that the regulatory authorities have the technical capabilities to move away for "files" (memory stick mentality) and move to databases/repositories/...

**Comment by swhume** [comment]
instead of pushing the data into an FDA API / repository, should the FDA GET study datasets from a repository that is either hosted by the sponsor or a third-part? In this case, the FDA would pull the content over once the sponsor has given them the green flag to proceed.

**Comment by JozefAerts** [comment]
Yes, I think so, probably third party, that's what I mean with "neutral zone".
On the other side, pushing it to FDA could also mean that the FDA is allowed to use it.
It all depends on what the FDA wants ...
